### PR TITLE
Fix CIAM Authority Tests

### PR DESCRIPTION
**Pull request by fidelianawar**
## Proposed changes

This PR is to fix a realm issue in the CIAM authority (makes sure the realm is correctly extracted from the URL and remove any unnecessary characters).

Related PRs:
[MSAL CIAM Changes](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1682) 
[Identity Core CIAM Changes](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1227) 

[Related work item](https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20Apple/Backlog%20items)

## Type of change

- [ ] Feature work
- [X] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/validation/MSIDCIAMAuthority.m [hunk 1]
```diff
@@ -57,6 +57,7 @@ - (nullable instancetype)initWithURL:(nonnull NSURL *)url
             if ([self.class isAuthorityFormatValid:url context:context error:nil])
             {
                 _url = [NSURL URLWithString:[NSString stringWithFormat:@"https://%@/%@", [url msidHostWithPortIfNecessary], rawTenant]];
+                url = _url;
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDCIAMAuthority.m:60]: Why is this needed?

File: IdentityCore/tests/MSIDCIAMAuthorityTests.m [hunk 2]
```diff
@@ -96,18 +96,6 @@ - (void)testInitCIAMAuthority_withValidUrlAndGUID_shouldReturnNilError
     XCTAssertNil(error);
 }
 
-- (void)testInitCIAMAuthority_withValidTenant_shouldReturnNilError
```
[the remark this answers is not in the document]
Comment (fidelianawar) [reply IdentityCore/tests/MSIDCIAMAuthorityTests.m:99]: this test was removed because it was essentially a repeat of another test that also tested CIAM authority with valid Tenant (testInitCIAMAuthority_withValidUrl_shouldReturnNilError) so I figured it was repetitive and could be taken out.

**Review by oldalton — APPROVED** [review]
